### PR TITLE
 Respect order of Package-Requires dependencies in cask pkg-file

### PR DESCRIPTION
**Pull request by cpitclaudel**
This is a rough draft: I'm still having trouble running tests; comments welcome!

Connects to #328

### CODE REVIEW

**Comment by cpitclaudel** [comment]
I'd love help in understanding why that test is failing, too; it's not clear to me what part of the changes should affect it.

**Comment by lunaryorn** [comment]
Well, changing the order of dependencies also changes the order in which immediate dependencies are installed, and thus also the order of packages in `package-alist` which tracks installed packages.  That's why the test is failing.

**Comment by cpitclaudel** [comment]
Is that the case? I felt that that test was simpler than this; does it involve dependencies?

**Comment by lunaryorn** [comment]
@cpitclaudel I think so.  Take a look at the test: It compiles a bundle with two direct dependencies (i.e. `depends-on`) and the _installs_ the bundle.

**Comment by rejeep** [comment]
Instead of changing code in many places, why not update [the source](https://github.com/cask/cask/blob/master/cask.el#L451)?

**Comment by cpitclaudel** [comment]
@rejeep I don't understand: that function isn't called if `deep` is `nil`.

**Comment by rejeep** [comment]
Maybe I was mistaken, hard to read code on a mobile... 😄 Will check again later

**Comment by cpitclaudel** [comment]
Thanks :)

File: cask.el [hunk 1]
```diff
@@ -556,7 +561,8 @@ This function return a `cask-bundle' object."
   "Initialize packages under PROJECT-PATH or `user-emacs-directory'.
 
 This function return a `cask-bundle' object."
-  (let* ((bundle (cask-setup (or project-path user-emacs-directory)))
+  (let* ((bundle
+          (cask-setup (or project-path user-emacs-directory)))
```
Comment (rejeep) [inline cask.el]: Did anything change here?

File: cask.el [hunk 2]
```diff
@@ -402,8 +405,7 @@ SCOPE may be nil or 'development."
          (cl-destructuring-bind (_ &rest patterns) form
            (setf (cask-bundle-patterns bundle) patterns)))
         (development
-         (cl-destructuring-bind (_ . body) form
-           (cask--eval bundle body 'development)))
+         (cask--eval bundle (cdr form) 'development))
```
Comment (rejeep) [inline cask.el]: Not related to this, right? In such case, please remove and make a separate pull request. We want to change as little as possible at a time.

**Comment by rejeep** [comment]
Also, take a look at the tests, like @lunaryorn mentioned above.
